pulp_ansible/app/tasks/utils.py

```diff
@@ -39,3 +43,57 @@ def filter_namespace(metadata, url):
         metadata["results"] = results
 
     return metadata
+
+
+def parse_collections_requirements_file(requirements_file_string):
+    """
+    Parses an Ansible requirement.yml file and returns all the collections defined in it.
+
+    The requirements file is in the form:
+        ---
+        collections:
+        - namespace.collection
+        - name: namespace.collection
+            version: version identifier, multiple identifiers are separated by ','
+            source: the URL or prededefined source name in ~/.ansible_galaxy
+                    to pull the collection from
+    :param requirements_file_string: The string of the requirements file.
+    :return: A list of tuples (name, version, source).
```

Comment:
Also can a raises section be added here to call out the non-standard Exception types this raises?